src/eckit/geo/Point.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * (C) Copyright 1996- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation nor
+ * does it submit to any jurisdiction.
+ */
+
+
+#pragma once
+
+#include <iosfwd>
+#include <variant>
+
+#include "eckit/geo/Point2.h"
+#include "eckit/geo/Point3.h"
+#include "eckit/geo/PointLonLat.h"
+
+
+namespace eckit::geo {
+
+
+using Point = std::variant<PointLonLat, Point2, Point3>;
```

Comment:
Note that `sizeof(Point) = 2 * sizeof(PointLonLat)`
(also an extra size_t for the internal variant index)
This means double! the memory if this were used in containers for "lon,lat".
Something that I'd have to be very careful about.

Reply:
This is by design -- Point is the generic argument for the Iterator and the Projection, so that we can explit polymorphism, chain projections and having the same interface for handling cartesian [X, Y, Z], cartesian [m], and geodetic coordinates without exploding the number of combinations in the interface. Note that the cost is only paid at construction (of each point), but it should be resuable as a temporary when used in an Iterator.

The usage in the Grid::to_points is another case wheer the penalty is fully paid, but that's intended for conversion to unstructured grid (which should not happen in or within performance-critical code, like a model) and interfacing with Python (or something else)

Reply:
Perhaps it is sufficient for `Grid::to_points` to return Point2 ?

Reply:
This way of using a std::variant I can attach a projection to a "usual" iterator converting (lon,lat) to (x,y,z) (or, but I don't have an example, to do (x,y) to (x,y,z)) to build a k-d tree in a very compact way, and the number of temporaries is minimal

Reply:
My commend there is that the specific resulting type depends on the configured Projection and Iterator for the containing Grid, so it is by design